recipes/full_finetune_distributed.py

```diff
@@ -545,6 +549,11 @@ def _setup_model(
         if self._compile:
             training.compile_model(model, verbose=self._is_rank_zero)
 
+        if self._enable_fp8_training:
+            # TODO: gate on nightlies?
```

Comment:
p.s. this is the PR in PT core I was talking about: https://github.com/pytorch/pytorch/pull/148922

if the PT core checkout has this PR, we don't need `Float8LinearConfig. force_recompute_fp8_weight_in_bwd` anymore as the min-cut partitioner now does the right thing automatically.